com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ActivitiExceptionEventHandler.java

```diff
@@ -0,0 +1,85 @@
+package com.sap.cloud.lm.sl.cf.process.util;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.sql.Timestamp;
+import java.text.MessageFormat;
+
+import org.activiti.engine.delegate.event.ActivitiEvent;
+import org.activiti.engine.delegate.event.ActivitiExceptionEvent;
+import org.activiti.engine.history.HistoricVariableInstance;
+import org.activiti.engine.impl.persistence.entity.VariableInstance;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sap.cloud.lm.sl.cf.process.message.Messages;
+import com.sap.cloud.lm.sl.common.SLException;
+import com.sap.cloud.lm.sl.persistence.model.ProgressMessage;
+import com.sap.cloud.lm.sl.persistence.model.ProgressMessage.ProgressMessageType;
+import com.sap.cloud.lm.sl.persistence.services.ProgressMessageService;
+
+public class ActivitiExceptionEventHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ActivitiExceptionEventHandler.class);
+
+    private ProgressMessageService progressMessageService;
+
+    public ActivitiExceptionEventHandler(ProgressMessageService progressMessageService) {
+        this.progressMessageService = progressMessageService;
+    }
+
+    public void handle(ActivitiEvent event) {
+        if (!(event instanceof ActivitiExceptionEvent)) {
+            return;
+        }
+
+        ActivitiExceptionEvent activitiExceptionEvent = (ActivitiExceptionEvent) event;
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        PrintStream printStream = new PrintStream(byteArrayOutputStream, true);

```

Comment:
The above could be done via ExceptionUtils::getRootCauseMessage or ExceptionUtils::getStackTrace